drivers/ethernet/eth_mcux.c

```diff
@@ -1029,14 +1029,18 @@ static int eth_init(const struct device *dev)
 
 #if defined(CONFIG_NET_IPV6)
```

Comment:
looks like in driver level this feature is not enabled for ipv4. see below, and above macro is limited to ipv6
```
static void eth_iface_init(struct net_if *iface)
{
	const struct device *dev = net_if_get_device(iface);
	struct eth_context *context = dev->data;

#if defined(CONFIG_NET_IPV6)
	static struct net_if_mcast_monitor mon;

	net_if_mcast_mon_register(&mon, iface, net_if_mcast_cb);
#endif /* CONFIG_NET_IPV6 */
```

Reply:
As for the drivers that are currently making use of the mcast monitor, I just tried to make sure that their functionality is preserved.